internal/dataplane/parser/translate_ingress.go

```diff
@@ -218,135 +216,204 @@ func (p *Parser) ingressRulesFromIngressV1() ingressRules {
-					objectSuccessfullyParsed = true
-				}
+	return p.ingressV1ToKongServiceLegacy(ingress, icp, servicesCache)
+}
+
+// ingressV1ToKongServiceLegacy translates IngressV1 object into Kong Service. It inserts the Kong Service into the passed servicesCache.
+// Returns true if the passed servicesCache was updated. It is used when CombinedRoutes feature flag is enabled.
+func (p *Parser) ingressV1ToKongServiceCombinedRoutes(
+	ingress *netv1.Ingress,
+	icp v1alpha1.IngressClassParametersSpec,
+	servicesCache kongServicesCache,
+) bool {
+	wasServicesCacheUpdated := false
+
+	regexPrefix := translators.ControllerPathRegexPrefix
+	if prefix, ok := ingress.ObjectMeta.Annotations[annotations.AnnotationPrefix+annotations.RegexPrefixKey]; ok {
+		regexPrefix = prefix
+	}
+	for _, kongStateService := range translators.TranslateIngress(ingress, p.flagEnabledRegexPathPrefix) {
+		for _, route := range kongStateService.Routes {
+			for i, path := range route.Paths {
+				newPath := translators.MaybePrependRegexPrefix(*path, regexPrefix, icp.EnableLegacyRegexDetection && p.flagEnabledRegexPathPrefix)
+				route.Paths[i] = &newPath
 			}
 		}
 
-		if objectSuccessfullyParsed {
-			p.ReportKubernetesObjectUpdate(ingress)
+		servicesCache[*kongStateService.Service.Name] = *kongStateService
+		wasServicesCacheUpdated = true
+	}
+
+	return wasServicesCacheUpdated
+}
+
+// ingressV1ToKongServiceLegacy translates IngressV1 object into Kong Service. It inserts the Kong Service into the passed servicesCache.
+// Returns true if the passed servicesCache was updated. It is used when the CombinedRoutes feature flag is disabled.
+func (p *Parser) ingressV1ToKongServiceLegacy(
+	ingress *netv1.Ingress,
+	icp v1alpha1.IngressClassParametersSpec,
+	servicesCache kongServicesCache,
+) bool {
+	wasServicesCacheUpdated := false
+
+	ingressSpec := ingress.Spec
+	maybePrependRegexPrefixFn := translators.MaybePrependRegexPrefixForIngressV1Fn(ingress, icp.EnableLegacyRegexDetection && p.flagEnabledRegexPathPrefix)
+	for i, rule := range ingressSpec.Rules {
+		if rule.HTTP == nil {
+			continue
+		}
+		for j, rulePath := range rule.HTTP.Paths {
+			pathTypeImplementationSpecific := netv1.PathTypeImplementationSpecific
+			if rulePath.PathType == nil {
+				rulePath.PathType = &pathTypeImplementationSpecific
+			}
+
+			paths := translators.PathsFromIngressPaths(rulePath, p.flagEnabledRegexPathPrefix)
+			if paths == nil {
+				// registering a failure, but technically it should never happen thanks to Kubernetes API validations
+				p.registerTranslationFailure(
+					fmt.Sprintf("could not translate Ingress Path %s to Kong paths", rulePath.Path), ingress,
+				)
+				continue
+			}
+
+			for i, path := range paths {
+				paths[i] = maybePrependRegexPrefixFn(*path)
+			}
+
+			r := kongstate.Route{
+				Ingress: util.FromK8sObject(ingress),
+				Route: kong.Route{
+					Name:              kong.String(fmt.Sprintf("%s.%s.%d%d", ingress.Namespace, ingress.Name, i, j)),
```

Comment:
`%d%d`? Is that expected? 

So there won't be any distinction between `i=1`,`j=11` `i=11`,`j=1`? 🤔 ?

Don't we want `%d.%d`?

Reply:
I think that your argument is valid, although I just wanted to move the code around with no changes to the behavior. It's how it was originally implemented and if we think that should be fixed, that's a separate story.

Reply:
I've created #3942 this to track/discuss this